conf/scalardb-custom-values.yaml

```diff
@@ -29,11 +35,26 @@ envoy:
                 operator: In
                 values:
                   - scalardbpool
+    podAntiAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+        - podAffinityTerm:
+            labelSelector:
+              matchExpressions:
+                - key: app.kubernetes.io/name
+                  operator: In
+                  values:
+                    - scalardb-server
```

Comment:
```suggestion
                    - scalardb
```
I think the Pod Anti Affinity does not work well if we set `scalardb-server` as the value of `app.kubernetes.io/name`.
To make Pod Anti Affinity work well, we need to set properly label information as the value of `matchExpressions`.

And the actual label of pods are the following.
```console
$ kubectl get pod scalardb-859cd45998-mlp6d -o jsonpath='{.metadata.labels}' | jq .
{
  "app.kubernetes.io/app": "scalardb",
  "app.kubernetes.io/instance": "scalardb",
  "app.kubernetes.io/name": "scalardb",
  "pod-template-hash": "859cd45998"
}
```
```console
$ kubectl get pod scalardb-envoy-585655996d-dfrp4 -o jsonpath='{.metadata.labels}' | jq .
{
  "app.kubernetes.io/app": "envoy",
  "app.kubernetes.io/instance": "scalardb",
  "app.kubernetes.io/name": "scalardb",
  "pod-template-hash": "585655996d"
}
```
The value of `app.kubernetes.io/name` is `scalardb`.
This value is decided based on the `chart name`.
https://github.com/scalar-labs/helm-charts/blob/main/charts/scalardb/templates/_helpers.tpl#L50
https://github.com/scalar-labs/helm-charts/blob/main/charts/scalardb/templates/_helpers.tpl#L5-L7

In other words, this values is fixed (hard-coded) as `scalardb`.
So, we need to set `scalardb` as the value of `app.kubernetes.io/name`.

---

For your reference, we can see the behavior when we set `scalardb` and `scalardb-server` as follows.

Note: I use `requiredDuringSchedulingIgnoredDuringExecution` instead of `preferredDuringSchedulingIgnoredDuringExecution` for the testing of label values since the `preferredDuringSchedulingIgnoredDuringExecution` does not force the rules. This setting is for label testing purposes only.

### Use `scalardb-server`

```yaml
scalardb:
  podAntiAffinity:
    requiredDuringSchedulingIgnoredDuringExecution:
    # preferredDuringSchedulingIgnoredDuringExecution:
    #   - podAffinityTerm:
        - labelSelector:
            matchExpressions:
              - key: app.kubernetes.io/name
                operator: In
                values:
                  - scalardb-server
              - key: app.kubernetes.io/app
                operator: In
                values:
                  - scalardb

envoy:
  podAntiAffinity:
    requiredDuringSchedulingIgnoredDuringExecution:
    # preferredDuringSchedulingIgnoredDuringExecution:
    #   - podAffinityTerm:
        - labelSelector:
            matchExpressions:
              - key: app.kubernetes.io/name
                operator: In
                values:
                  - scalardb-server
              - key: app.kubernetes.io/app
                operator: In
                values:
                  - envoy
```
```console
$ kubectl get node
NAME       STATUS   ROLES                  AGE   VERSION
minikube   Ready    control-plane,master   34d   v1.23.3
```
```console
$ kubectl get pod -o wide
NAME                              READY   STATUS    RESTARTS   AGE     IP            NODE       NOMINATED NODE   READINESS GATES
postgresql-scalardb-0             1/1     Running   0          3h42m   172.17.0.8    minikube   <none>           <none>
scalardb-859cd45998-2mbtg         1/1     Running   0          117s    172.17.0.13   minikube   <none>           <none>
scalardb-859cd45998-brbgd         1/1     Running   0          11m     172.17.0.10   minikube   <none>           <none>
scalardb-859cd45998-lz8jc         1/1     Running   0          117s    172.17.0.14   minikube   <none>           <none>
scalardb-envoy-585655996d-5dsxn   1/1     Running   0          117s    172.17.0.12   minikube   <none>           <none>
scalardb-envoy-585655996d-rqtps   1/1     Running   0          11m     172.17.0.9    minikube   <none>           <none>
scalardb-envoy-585655996d-sbjln   1/1     Running   0          117s    172.17.0.11   minikube   <none>           <none>
```
You can see the three pods are running on the same node (Pod Anti Affinity is not working).

### Use `scalardb`

```yaml
scalardb:
  podAntiAffinity:
    requiredDuringSchedulingIgnoredDuringExecution:
    # preferredDuringSchedulingIgnoredDuringExecution:
    #   - podAffinityTerm:
        - labelSelector:
            matchExpressions:
              - key: app.kubernetes.io/name
                operator: In
                values:
                  - scalardb
              - key: app.kubernetes.io/app
                operator: In
                values:
                  - scalardb

envoy:
  podAntiAffinity:
    requiredDuringSchedulingIgnoredDuringExecution:
    # preferredDuringSchedulingIgnoredDuringExecution:
    #   - podAffinityTerm:
        - labelSelector:
            matchExpressions:
              - key: app.kubernetes.io/name
                operator: In
                values:
                  - scalardb
              - key: app.kubernetes.io/app
                operator: In
                values:
                  - envoy
```
```console
$ kubectl get node
NAME       STATUS   ROLES                  AGE   VERSION
minikube   Ready    control-plane,master   34d   v1.23.3
```
```console
$ kubectl get pod -o wide
NAME                              READY   STATUS    RESTARTS   AGE     IP            NODE       NOMINATED NODE   READINESS GATES
postgresql-scalardb-0             1/1     Running   0          3h46m   172.17.0.8    minikube   <none>           <none>
scalardb-66d98569b4-vb8x6         0/1     Pending   0          38s     <none>        <none>     <none>           <none>
scalardb-66d98569b4-w2lh9         0/1     Pending   0          38s     <none>        <none>     <none>           <none>
scalardb-66d98569b4-x2t24         1/1     Running   0          38s     172.17.0.10   minikube   <none>           <none>
scalardb-envoy-66dc669776-5vpqf   0/1     Pending   0          38s     <none>        <none>     <none>           <none>
scalardb-envoy-66dc669776-5zknb   0/1     Pending   0          38s     <none>        <none>     <none>           <none>
scalardb-envoy-66dc669776-8tjl6   1/1     Running   0          38s     172.17.0.9    minikube   <none>           <none>
```
You can see the several pods are pending (Pod Anti Affinity is working).

Reply:
@kota2and3kan san,
Thank you for the detailed explanation. 
I understood it more clearly.
I have updated it as you suggested.

Reply:
@inv-jishnu Please check and test what you updated. It's not scalable for reviewers to check it.